### DISCUSSION TITLE
Advance Blocking - reload over CLI

### DISCUSSION THREAD

**Discussion by PeterKnotek** [TechnitiumSoftware/DnsServer#904] [General]
I'm thinking about a bash script for variable IP addition for blocking www pages at the DNS level
This would probably not be a problem
Does advance blocking CLI command for reload CFG ???

Thanks very much

**Comment by ShreyasZare** [comment]
Thanks for the post. There is no CLI option available. You will need to use the HTTP API in your bash script with curl and jq commands.

**Reply by PeterKnotek** [reply]
can you tell me .... which API command should I use to reload CFG for advanced blocking ...
Thanks in advance

**Reply by ShreyasZare** [reply]
You can do that by first calling the [Get App Config](https://github.com/TechnitiumSoftware/DnsServer/blob/master/APIDOCS.md#get-app-config) API and then modify the json data it returns and post it to the [Set App Config](https://github.com/TechnitiumSoftware/DnsServer/blob/master/APIDOCS.md#set-app-config) API. This will immediately reload the new config.

**Reply by PeterKnotek** [reply]
HI - thanks for your response, but what is the exact name for Advances Blocking app ?? Thanks  very much

**Reply by ShreyasZare** [reply]
The name of the app is the one you get in the [List Apps](https://github.com/TechnitiumSoftware/DnsServer/blob/master/APIDOCS.md#list-apps) API call. Which is actually just the name you see in the admin web panel.

**Reply by PeterKnotek** [reply]
Thanks  - but stil not wok wit name advancedblocking
here is my api call for apps - picture
![save](https://github.com/TechnitiumSoftware/DnsServer/assets/101987299/0005cfdb-a6ae-4b67-af8b-3d59b267b819)
if call :  http://localhost:5380/api/apps/config/get?token=x&name=app-name, where appname is "Advanced Blocking" or AdvancedBlocking ... I tried different combinations - not show
Thanks

**Reply by ShreyasZare** [reply]
The name should be "Advanced Blocking" exactly and in the URL parameter, it needs to be url encoded so the actual value in the URL should be "Advanced%20Blocking". If it still does not work then share the exact URL that you are calling and the exact response it returns.

**Reply by PeterKnotek** [reply]
HI
Call  http://localhost:5380/api/apps/config/get?token=x&name="AdvancedBlocking" - stil return error
![apps_api](https://github.com/TechnitiumSoftware/DnsServer/assets/101987299/ec995891-c567-4b9d-84ed-7777d0f7c47f)

Try to simulate it  please and write what the exact URL :D
Thanks in advance

**Reply by ShreyasZare** [reply]
You missed adding the space character "%20" and also there should be no double quotes.

Try this exact URL with your token: `http://localhost:5380/api/apps/config/get?token=x&name=Advanced%20Blocking`

**Reply by PeterKnotek** [reply]
Thanks
name=Advanced%20Blocking - work 
But .... how reload CFG for Advanced Blocking ( only for reload )
- import cfg to /etc/dns/apps/Advanced\ Blocking/dnsApp.config
  - import with  bash script and then reload with api command
Thanks

**Reply by ShreyasZare** [reply]
It may be a bit confusing but, there is no "config reload" operation available. The API only supports "get" and "set" operations for the config. Also, its not recommended to use/edit the config on disk since the software is designed only to be configurable via API. So, you need to use the API to get the existing config, add your changes, and then use the API to set the config. The set action will load the new changes instantly.

**Comment by ravor-org** [comment]
Time to revive this one:
I'm currently having the same problem, that I have local blocklists (served by an internal webserver) and want to force a refresh of the lists. How do I do that? Is there an API request to refresh/redownload the lists?

**Reply by ShreyasZare** [reply]
Thanks for asking. There is no option to do that now directly. You will need to remove the block list URL from the config once and save it. You can use JSON comments to comment existing URL out. Then again edit the config to add the URL back to do a force reload.

**Reply by ravor-org** [reply]
Any chance to get this implemented in API? I would think, that this is use case, that's used more often by power users?

**Reply by ShreyasZare** [reply]
The DNS apps do not support APIs. So this will mean to add a new functionality for DNS apps framework which is not something that can be done quick enough.
